src/main/java/org/cirdles/topsoil/app/TopsoilMainWindow.java

```diff
@@ -116,12 +116,18 @@ public void initialize(URL url, ResourceBundle resources) {
 
     Optional<TSVTable> getCurrentTable() {
         Optional<TSVTable> result = Optional.empty();
-        Tab currentTab = dataTableTabPane.getSelectionModel().getSelectedItem();
+        Tab currentTab = getCurrentTab();
+
+        if(currentTab==null)
+        {
+            createTab();
+            currentTab = getCurrentTab();
+        }
```

Comment:
With some refactoring, the above is better written as:

``` java
Optional<Tab> currentTab = Optional.ofNullable(getCurrentTab()).orElse(createTab());
```

Of course, this might not belong here, but generally, `Optional` should be used whenever you itch to write `if (currentTab == null)`.